enhancements/installer/azure-internal-with-user-defined-routing.md

```diff
@@ -0,0 +1,243 @@
+
+For outbound type UserDefined the installer does not create the dummy inbound_rule or the dummy Kubernetes service type Loadbalancer. The installer also configures the kubernetes cloud controller to disable outbound egress for all public Kubernetes service type Loadbalancers.
+
+### User Defined routing requirements
+
+Here are some of the expectations from the outbound routing setup by users,
+
+- Users must ensure that egress to Internet is possible to pull container images unless using a internal registry mirror.
+- Users must ensure that the cluster is able to access the Azure APIs from the cluster.
+
+### User Stories
+
+#### Internal cluster with PROXY for Internet access
+
+The user will be using proxy configuration with user defined routing to allow egress to internet. Users must be careful that cluster operators currently do not access the Azure APIs using proxy and therefore access to Azure APIs should be available outside of proxy.
+
+When using the default route table for subnets with 0.0.0.0/0 populated automatically by Azure, all Azure APIs requests are routed over Azure's internal network even though the IP addresses are public. So in this scenario as long as the Network Security Group rules allow egress to Azure API endpoints, proxy with user defined routing configuration in install-config.yaml will allow users to create internal clusters with no public endpoints.
+
+#### Internal cluster with Azure Firewall for Internet access
+
+Users can use Azure Firewall to provide outbound routing for the virtual network used to install the cluster. An example of such a setup is outlined [here](https://docs.microsoft.com/en-us/azure/aks/egress-outboundtype#deploy-a-cluster-with-outbound-type-of-udr-and-azure-firewall)
+
+Using a virtual network setup as outlined above and user defined routing conifiguration the users can create internal clusters with no public endpoints
+
+#### Internal cluster with no Internet access
+
+Users that have virtual networks with no access to Internet depend on internal registry mirrors accesible to the private network to pull container images. So as long as the access to the Azure APIs is ensured from the cluster, the users can use the imageContentSources and user defined outbound type routing to create internal clusters with no public endpoints.
+
+### Risks and Mitigations
+
+1. There are no great ways to validate if the user's outbound routing setup is configured correctly for the OpenShift cluster and therefore it will result in not easy to debug failure modes and will probably require a lot more back and forth to understand the setup to triage the issue. The bootstrap failure log bundle is based on SSH access to bootstrap machines and therefore those logs should help uncover the symptons to atleast narrow down the failures.
```

Comment:
+1.  I think that one possible validation may be checking that the vnet has a route table attached and that the route table contains a 0.0.0.0/0 route?